components/pdf_preview.jsx

```diff
@@ -40,14 +38,20 @@ export default class PDFPreview extends React.PureComponent {
             loading: true,
             success: false,
         };
-
-        for (let i = 0; i < MAX_PDF_PAGES; i++) {
-            this[`pdfCanvasRef-${i}`] = React.createRef();
-        }
     }
 
     componentDidMount() {
         this.getPdfDocument();
+        this.parentNode = document.querySelector('.modal .modal-image .modal-image__content');
```

Comment:
This is far from be optimal, a workaround for this is to pass the scroll viewport information (top and bottom) from the parent component.

Reply:
This is needed because the parent div has `this.container` scroll within it, right? I think ideally we'd make this component scroll, but I'm fine doing it this way since the rest of those modals needs work anyway.

Could we possibly get the parent by calling `this.container.parentNode` instead of using the query selector? That might be a bit less fragile

Reply:
Improved it.